testsuite/MDAnalysisTests/coordinates/test_trc.py

```diff
@@ -0,0 +1,189 @@
+        # first frame first particle
+        TRC_U.trajectory[0]
+        assert_allclose(TRC_U.atoms.positions[0],
+                        [2.19782507, 24.65064345, 29.39783426])
+        # fith frame first particle
+        TRC_U.trajectory[4]
+        assert_allclose(TRC_U.atoms.positions[0],
+                        [0.37026654, 22.78805010, 3.69695262])
+
+    def test_trc_dimensions(self, TRC_U):
+        assert TRC_U.trajectory[0].dimensions is None
+
+    def test_trc_n_frames(self, TRC_U):
+        assert len(TRC_U.trajectory) == 6
+        assert (TRC_U.trajectory.n_frames) == 6
+
+    def test_trc_n_atoms(self, TRC_U):
+        assert (TRC_U.trajectory.n_atoms) == 73
+
+    def test_trc_frame(self, TRC_U):
+        assert TRC_U.trajectory[0].frame == 0
+        assert TRC_U.trajectory[4].frame == 4
+
+    def test_trc_time(self, TRC_U):
+        assert TRC_U.trajectory[0].time == 0
+        assert TRC_U.trajectory[4].time == 80
+
+    def test_trc_dt(self, TRC_U):
+        time_array = np.array([ts.time for ts in TRC_U.trajectory])
+        assert_equal(time_array, np.arange(6)*20)
+
+        dt_array = np.diff(time_array)
+        assert_equal(dt_array, np.full(5, 20))
+
+    def test_trc_data_step(self, TRC_U):
+        assert TRC_U.trajectory[0].data['step'] == 0
+        assert TRC_U.trajectory[4].data['step'] == 10000
+
+    def test_periodic(self, TRC_U):
+        assert_equal(TRC_U.trajectory.periodic, False)
+
+    def test_rewind(self, TRC_U):
+        TRC_U.trajectory[0]
+        trc = TRC_U.trajectory
+        trc.next()
+        trc.next()
+        trc.next()
+        trc.next()
+        assert_equal(trc.ts.frame, 4,
+                     "trajectory.next() did not forward to frameindex 4")
+        trc.rewind()
+        assert_equal(trc.ts.frame, 0, "trajectory.rewind() failed "
+                                      "to rewind to first frame")
+
+        assert np.any(TRC_U.atoms.positions != 0), "The atom positions " \
+                                                   "are not populated"
+
+    def test_random_access(self, TRC_U):
+        TRC_U.trajectory[0]
+        pos0 = TRC_U.atoms[0].position
+        TRC_U.trajectory.next()
+        TRC_U.trajectory.next()
+        pos2 = TRC_U.atoms[0].position
+
+        TRC_U.trajectory[0]
+
+        assert_equal(TRC_U.atoms[0].position, pos0)
+
+        TRC_U.trajectory[2]
+
+        assert_equal(TRC_U.atoms[0].position, pos2)
+
+    def test_read_frame_reopens(self, TRC_U):
+        TRC_U.trajectory._reopen()
+        TRC_U.trajectory[4]
+        assert TRC_U.trajectory.ts.frame == 4
+
+
+class TestTRCReaderSolvatedBox:
+    @pytest.fixture(scope='class')
+    def TRC_U(self):
+        return mda.Universe(TRC_PDB_SOLV, TRC_TRAJ_SOLV)
+
+    def test_trc_n_atoms(self, TRC_U):
+        assert (TRC_U.trajectory.n_atoms) == 2797
+
+    def test_periodic(self, TRC_U):
+        assert_equal(TRC_U.trajectory.periodic, True)
+
+    def test_trc_dimensions(self, TRC_U):
```

Comment:
If gromos11 supports triclinic boxes, can we also test that? May need a new file.

Reply:
Yes, I don't understand if the reader supports general triclinic boxes because it raises an exception for "truncated octahedron PBC" although below you have a test for triclinic boxes. Are the "truncated octahedron PBC" something special in GROMOS?

Reply:
I added support for triclinic boxes as well as some tests with my last commits yesterday. 

The gromos program supports four kinds of boundary conditions:
- truncated-octahedron (-1)
- vacuum (0)
- rectangular (1)
- triclinic (2)

Even through the truncated-octahedron can be converted into a triclinic box with gromos-tools, it is regarded as its own boundary conditions. I don't know if mdanalysis supports truncated-octahedral PBC, but I didn't want to do some internal conversions hidden from the user. Therefore, PBC option -1 (truncated-octahedron) is not supported while 0,1 and 2 are.

Reply:
We don't explicitly support truncated octahedrals.  I think it would be appropriate to silently convert this to the equivalent triclinic box.  Elsewhere on the tracker there's some discussion about also storing the canonical/original form of the box from the trajectory (the other case was where the triclinic box was defined as a non-diagonally sparse representation, but this seems like a variation on that same idea)

Reply:
This could be done in a future PR though.  I think in general conversion of octahedrals to triclinics would be a neat addition

Reply:
I am ok with a future PR to enhance the functionality for MDA to internally convert GROMOS special PBCs to generic ones. Would be best to raise a PR so that it doesn't get forgotten.